### PR TITLE
Add CkReferenceMsg function

### PR DESCRIPTION
**Pull request by rbuch**
A common paradigm in the codebase is to increment the refcount of a
message by using CmiReference(UsrToEnv(msg)), this creates a function
to simplify this usage.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
Why does this return the message?

**Comment by stwhite91** [comment]
> Why does this return the message?

That keeps the function signature the same as CkCopyMsg() and can be useful, as in PR #2861

**Comment by epmikida** [comment]
> May be a good idea to have a PR which cleans up usage where relevant now that we have this. A search/replace for `CmiReference(UsrToEnv(*))` -> `CkReferenceMsg(*)` shouldn't be too hard to do.

Nevermind...you already did that. I guess I was just expecting more instances of it.

File: src/ck-core/debug-charm.C [hunk 1]
```diff
@@ -67,7 +67,7 @@ void CpdBeforeEp(int ep, void *obj, void *msg) {
     entry.obj = obj;
     if (msg != NULL) {
       entry.msg = msg;
```
Comment (stwhite91) [inline src/ck-core/debug-charm.C:69]: The body of this if-statement can be simplified to 'entry.msg = CkReferenceMsg(msg);' now

File: src/xlat-i/xi-Parameter.C [hunk 2]
```diff
@@ -712,7 +712,7 @@ void ParamList::beginUnmarshallSDAGCall(XStr& str, bool usesImplBuf) {
     if (hasArray || hasRdma()) {
       if (!usesImplBuf) {
         str << "  genClosure->_impl_marshall = impl_msg_typed;\n";
```
Comment (stwhite91) [inline src/xlat-i/xi-Parameter.C:714]: These two lines could be combined as well

File: src/ck-core/debug-charm.C (hunk 1 above)
Comment (rbuch) [reply src/ck-core/debug-charm.C:69]: Good call, done.

File: src/xlat-i/xi-Parameter.C (hunk 2 above)
Comment (rbuch) [reply src/xlat-i/xi-Parameter.C:714]: Doing that causes a type error due to it returning a `void*`, so I'm leaving this one as it is.

**Pull request merged**